src/core/lib/slice/slice_buffer.c

```diff
@@ -61,12 +56,15 @@ static void maybe_embiggen(grpc_slice_buffer *sb) {
     if (sb->base_slices == sb->inlined) {
       sb->base_slices = gpr_malloc(sb->capacity * sizeof(grpc_slice));
       memcpy(sb->base_slices, sb->inlined, slice_count * sizeof(grpc_slice));
+      sb->slices = sb->base_slices + slice_offset;
     } else {
+      if (sb->base_slices != sb->slices) {
```

Comment:
It would be good to do this *before* growing the capacity: i.e line 54 above  `sb->capacity = GROW(sb->capacity)` 
Because after the `memmove`, we might have some space and no longer have to do a `gpr_realloc`

So perhaps it would be just good do the memmove just before checking for `if (slice_count == sb->capacity)`..

i.e something like:
``` C
 static void maybe_embiggen(grpc_slice_buffer *sb) {
    size_t slice_offset = (size_t)(sb->slices - sb->base_slices);
    size_t slice_count = sb->count + slice_offset;
  
    if (slice_count == sb->capacity && sb->base_slices != sb->slices) {
      /* do the memmove */
      /* re-calculate slice_offset or rather just reset slice_count to sb->count */
    }
    ..
}
```